internal/aws/awsutil/conn.go

```diff
@@ -37,14 +41,22 @@ import (
 
 type ConnAttr interface {
 	newAWSSession(logger *zap.Logger, roleArn string, region string, profile string, sharedCredentialsFile []string) (*session.Session, error)
-	getEC2Region(s *session.Session) (string, error)
+	getEC2Region(cfg awsSDKV2.Config) (string, error)
 }
 
 // Conn implements connAttr interface.
 type Conn struct{}
 
-func (c *Conn) getEC2Region(s *session.Session) (string, error) {
-	return ec2metadata.New(s).Region()
+func (c *Conn) getEC2Region(cfg awsSDKV2.Config) (string, error) {
```

Comment:
It looks like `GetAWSConfig()` will set `cfg.Region = region`.
So couldn't `getEC2Region()` just return `cfg.Region`?